container/lxd/lxd.go

```diff
@@ -123,7 +123,9 @@ func (manager *containerManager) CreateContainer(
 		return nil, nil, errors.Trace(err)
 	}
 
-	userData, err := containerinit.CloudInitUserData(instanceConfig, networkConfig)
+	// Do not pass networkConfig, as we want to directly inject our own ENI
+	// rather than using cloud-init.
+	userData, err := containerinit.CloudInitUserData(instanceConfig, nil)
 	if err != nil {
 		return
```

Comment:
I *think* you've changed err scope here because it is used in := statement. Please provide an explicit return as above with errors.Trace \o/